lib/signature.dart

```diff
@@ -196,17 +201,88 @@ class SignatureController extends ValueNotifier<List<Point>> {
     }
     _SignaturePainter(this).paint(canvas, null);
     var picture = recorder.endRecording();
-    return picture.toImage(
-        (maxX - minX + penStrokeWidth * 2).toInt(), (maxY - minY + penStrokeWidth * 2).toInt());
+    return picture.toImage((maxX - minX + penStrokeWidth * 2).toInt(),
+        (maxY - minY + penStrokeWidth * 2).toInt());
   }
 
   Future<Uint8List> toPngBytes() async {
-    var image = await toImage();
-    if (image == null) {
-      return null;
+    try {
+      var image = await toImage();
+      if (image == null) {
+        return null;
+      }
+
+      var bytes = await image.toByteData(format: ui.ImageByteFormat.png);
+      return bytes.buffer.asUint8List();
+    } catch (e) {
+      return _toPngBytesForWeb();
```

Comment:
Using exceptions for flow control doesn't seem right for me. 

Better solution would be to check value of variable [kIsWeb](https://stackoverflow.com/questions/57937280/how-can-i-detect-if-my-flutter-app-is-running-in-the-web) from foundation.dart

Reply:
right! Done